src/roman_datamodels/datamodels/_datamodels.py

```diff
@@ -440,3 +441,74 @@ class ImageSourceCatalogModel(_RomanDataModel, _ParquetMixin):
 
 class SegmentationMapModel(_RomanDataModel):
     _node_type = stnode.SegmentationMap
+
+
+class WfiWcsModel(_RomanDataModel):
+    _node_type = stnode.WfiWcs
+
+    @classmethod
+    def from_model_with_wcs(cls, model, l1_border=4):
+        """Extract the WCS information from an exposure model post-assign_wcs
+
+        Construct a `WfiWcsModel` from any model that is used post-assign_wcs step
+        in the ELP pipeline. The WCS information is extracted out of the input model.
+        The wcs-related meta information is copied verbatim from the input model.
+
+        However, the WCS object itself is placed into the attribute 'wcs_l2'. Furthermore, a
+        modified GWCS, applicable to the Level 1 version of the input model, is created
+        and stored in the attribute 'wcs_l1'.
+
+        Parameters
+        ----------
+        model : ImageModel
+            The input data model (a WfiWcsModel will also work).
+
+        l1_border : int
+            The extra border to add for the L1 wcs.
+
+        Returns
+        -------
+        wfiwcs_model : WfiWcsModel
+            The WfiWcsModel built from the input model. If the input is already
+            a WfiWcsModel, it is simply returned.
+
+        """
+        ALLOWED_MODELS = ImageModel
+
+        if isinstance(model, cls):
+            return model
+        if not isinstance(model, ALLOWED_MODELS):
+            raise ValueError(f"Input must be one of {ALLOWED_MODELS}")
+
+        # Ensure a WCS has been defined.
+        if model.meta.wcs is None:
+            raise ValueError(f"Model has no WCS defined, cannot create {cls}")
+
+        # Create base node with dummy values (for validation)
+        from roman_datamodels.maker_utils import mk_wfi_wcs
+
+        wfi_wcs = mk_wfi_wcs()
+
+        _node_update(wfi_wcs, model, only_in_to_node=True)
```

Comment:
```suggestion
        wfi_wcs = cls()
        wfi_wcs.meta = {}
        skip_keys = {"calibration_software_version", "product_type", "filename", "file_date"}
        for k in wfi_wcs.meta._schema_attributes.explicit_properties:
            if k not in skip_keys and k in model.meta:
                wfi_wcs.meta[k] = copy.deepcopy(model.meta[k])
```
Generally yes there's no substitute for the maker_utils. In this case the above suggestion would copy over the schema-defined metadata from the input model (without introducing new fake values). It might be easier at this point to remove the `skip_keys` although it does highlight that having this conversion code in roman_datamodels doesn't allow access to the romancal version. Any reason to not have this conversion code in romancal?

Also, what should `product_type` be for this model?

Finally, as there's no real solution for `file_date` copying the junk value from the input seems like a reasonable option here. Perhaps the same is true for `filename` although `none` seems marginally better.

Reply:
Concerning the statement "conversion code in roman_datamodels doesn't allow access to the romancal version": What is meant by "version"? This is simply a conversion of one model to another and not sure what dependency on the any higher level package should be, whether it is romancal or something else?

But this does answer the question asked in slack about "is there a way of getting the tree", and it appears there is. I will explore that. However, this will involve updating of `_node_update` because there is other functionality needed. But if that can be traversed using the `_schema_attributes`, we'll take it.

Reply:
I see, the `_schema_attributes` only exists for `meta`. So top level attributes cannot be devined from this way. For this case, the above will work. Thanks!

Reply:
> Concerning the statement "conversion code in roman_datamodels doesn't allow access to the romancal version": What is meant by "version"? This is simply a conversion of one model to another and not sure what dependency on the any higher level package should be, whether it is romancal or something else?

"meta.calibration_software_version" records the romancal version number/string. This is the version I was referring to. It's required to be defined for a valid file and is typically set by `Step.finalize_result` in romancal. The point I was trying to make was that if this conversion code was in romancal it could fill in that version string.

> But this does answer the question asked in slack about "is there a way of getting the tree", and it appears there is. I will explore that. However, this will involve updating of `_node_update` because there is other functionality needed. But if that can be traversed using the `_schema_attributes`, we'll take it.
>
> I see, the _schema_attributes only exists for meta. So top level attributes cannot be devined from this way. For this case, the above will work. Thanks!

`_schema_attributes` exists for any `DNode`: `meta`, also `meta.exposure`, etc and even the top level node `WfiWcsNode` (or something like that) in this case. This doesn't fully define the "tree" as the schema can contain things like "patternProperties" which constrain a possible value but don't define an "explicit" value.

Reply:
will explore that bit later, but now just using the provided code.